### DISCUSSION TITLE
Creating a custom token for row access control

### DISCUSSION THREAD

**Discussion by Subh94** [hasura/graphql-engine#10525] [Question]
I created a auth.session.accessToken.customClaims and provide a valid path like 
key = 'Vendor-Id'
value = 'org.organization_id'

Where graphQL Query can resolve value on defined user and organization member but `X-Hasura-Vendor-Id` is not reflecting on graphql and throwing an error 

{
  "errors": [
    {
      "message": "missing session variable: \"x-hasura-vendor-id\"",
      "extensions": {
        "path": "$",
        "code": "not-found"
      }
    }
  ]
}

Anything I am missing

**Comment by manasag** [comment]
Hi, can you share more details: 
- `auth.session.accessToken.customClaims` is this referring to a JWT claim. Can you share full structure of this claims
- Share permission rule or wherever you are setting this variable check.

**Comment by Subh94** [comment]
<img width="1013" alt="image" src="https://github.com/user-attachments/assets/a98ab6ce-ae2b-4df3-94a3-e560021b0195">
Created a Permission Variable as above described on nhost https://docs.nhost.io/guides/api/permissions#permission-variables

Permission is like {"_or":[{"user_id":{"_eq":"X-Hasura-User-Id"}},{"organization_id":{"_eq":"X-Hasura-Vendor-Id"}}]}
in Hasura 

on table organization_id column exist.

**Comment by Subh94** [comment]
Sorry I posted on wrong discussion look like, Intended to post on nhost discussion.
